enhancements/dns/plugins.md

```diff
@@ -97,24 +98,28 @@ resolution, it restricts how cluster operators manage DNS name queries.
 
 ## Proposal
 
-`PluginConfig` defines one or more CoreDNS plugins that can be configured. If a plugin is not defined, it will use a
-default configuration provided by the cluster-dns-operator. A plugin is represented as a slice when multiple instances
-of the plugin can be expressed within the DNS configuration.
+`Server` defines the schema for a server that runs per instance of CoreDNS. A `Server` must have a `Name` that is unique
+among `Servers`. `Zones` defines one or more subdomains to match for name queries to be forwarded to `Upstreams`.
+`ForwardPlugin` provides options for configuring the forward plugin configuration. A `Server` listens on port `5353`,
+which can not be configured at this time. If a `Server` is not specified, the cluster-dns-operator will generate the
```

Comment:
I think the proposal would be clearer if this paragraph did not mention upstreams or the forward plugin.  The `name` field is required to identify the server, and `zones` is required to indicate for which domains the server is authoritative; these two fields are integral to the definition of a server.  However, the `forwardPlugin` field may in the future be one of many optional fields, so defining `zones` in terms of a specific plugin may confound matters now and in the future.

Reply:
I'm reorganizing this section based on your feedback.